sphinx/tutorial/cylc/furthertopics/family-triggers.rst

```diff
@@ -0,0 +1,184 @@
+will need to stop your suite using the "stop" button in the cylc gui in order
+to allow it to shutdown.
+
+
+Family triggering: success
+--------------------------
+
+As you will have noticed by watching the suite run, some of the tasks in the
+``MINERS`` family succeed and some fail.
+
+We would like to add a task to sell any diamonds we find, but wait for all
+the miners to report back first so we only make the one trip.
+
+We can address this by using *family triggers*. In particular, we are going
+to use the ``finish-all`` trigger to check for all members of the ``MINERS``
+family finishing, and the ``succeed-any`` trigger to check for any of the
+tasks in the ``MINERS`` family succeeding.
+
+Open your ``suite.rc`` file and change the ``[[dependencies]]`` to look like
+this:
+
+.. code-block:: cylc
+
+   [[dependencies]]
+       graph = """visit_mine => MINERS
+                  MINERS:finish-all & MINERS:succeed-any => sell_diamonds"""
+
+Then, add the following task to the ``[runtime]`` section:
+
+.. code-block:: cylc
+
+   [[sell_diamonds]]
+      script = sleep 5
+
+These changes add a ``sell_diamonds`` task to the suite which is run once
+all the ``MINERS`` tasks have finished and if any of them have succeeded.
+
+Save your changes and run your suite. You should see the new
+``sell_diamonds`` task being run once all the miners have finished and at
+least one of them has succeeded. As before, stop your suite using the "stop"
+button in the ``cylc gui``.
+
+
+Family triggering: failure
+--------------------------
+
+Cylc also allows us to trigger off failure of tasks in a particular family.
+
+We would like to add another task to close down unproductive mineshafts once
+all the miners have reported back and had time to discuss their findings.
+
+To do this we will make use of family triggers in a similar manner to before.
+
+Open your ``suite.rc`` file and change the ``[[dependencies]]`` to look like
+this:
+
+.. code-block:: cylc
+
+   [[dependencies]]
+       graph = """visit_mine => MINERS
+                  MINERS:finish-all & MINERS:succeed-any => sell_diamonds
+                  MINERS:finish-all & MINERS:fail-any => close_shafts
+                  close_shafts => !MINERS
+                  """
+
+Alter the ``[[sell_diamonds]]`` section to look like this:
+
+.. code-block:: cylc
+
+   [[close_shafts, sell_diamonds]]
+       script = sleep 5
+
+These changes add a ``close_shafts`` task which is run once all the
+``MINERS`` tasks have finished and any of them have failed. On completion
+it applies a *suicide trigger* to the ``MINERS`` family in order to allow
+the suite to shutdown.
+
+Save your changes and run your suite. You should see the new
+``close_shafts`` run should any of the ``MINERS`` tasks be in the failed
+state once they have all finished.
+
+
+Different triggers
+------------------
+
+Other types of triggers beyond those covered in the example are also available.
```

Comment:
Explicitly mention qualifiers and reference the 'Cylc Tutorial -> Scheduling -> Further Scheduling -> Qualifiers' section here given that these triggers are available for each type of standard qualifier.

Reply:
Changed "trigger" to "qualifier" (which makes more sense).